lib/improver/cli/temporal_interpolate.py

```diff
@@ -31,123 +31,43 @@
-                       "can be provided separated by a space "
-                       "but if times are set interval_in_mins "
-                       "can not be set")
-
-    parser.add_argument("--interpolation_method",
-                        metavar="INTERPOLATION_METHOD",
-                        default="linear",
-                        choices=["linear", "solar", "daynight"],
-                        help="Specifies the interpolation method; "
-                        "solar interpolates using the solar elevation, "
-                        "daynight uses linear interpolation but sets"
-                        " night time points to 0.0, "
-                        "linear is linear interpolation. "
-                        "Default is linear.")
-
-    parser.add_argument("--output_files", metavar="OUTPUT_FILES",
-                        required=True, nargs="+",
-                        help="List of output files."
-                        " The interpolated files will always be"
-                        " in the chronological order of"
-                        " earliest to latest "
-                        " regardless of the order of the infiles.")
-
-    args = parser.parse_args(args=argv)
-
-    # Load Cubes
-    cube_0 = load_cube(args.infiles[0])
-    cube_1 = load_cube(args.infiles[1])
-
-    # Process Cubes
-    interpolated_cubes = process(cube_0, cube_1, args.interval_in_mins,
-                                 args.times, args.interpolation_method)
-
-    # Save Cubes
-    len_files = len(args.output_files)
-    len_cubes = len(interpolated_cubes)
-    if len_files == len_cubes:
-        for i, cube_out in enumerate(interpolated_cubes):
-            save_netcdf(cube_out, args.output_files[i])
-    else:
-        msg = ("Output_files do not match cubes created. "
-               "{} files given but {} required.".format(len_files,
-                                                        len_cubes))
-        raise ValueError(msg)
-
-
-def process(cube_0, cube_1, interval_in_mins=None, in_times=None,
+@cli.clizefy
+@cli.with_output
+def process(start_cube: cli.inputcube,
+            end_cube: cli.inputcube,
+            *,
+            interval_in_mins: int = None,
+            times: cli.comma_separated_list = None,
             interpolation_method='linear'):
-    """Module to interpolate data between validity times.
+     """Interpolate data between validity times.
 
     Interpolate data to intermediate times between the validity times of two
     cubes. This can be used to fill in missing data (e.g. for radar fields)
     or to ensure data is available at the required intervals when model data
     is not available at these times.
 
     Args:
-        cube_0 (iris.cube.Cube):
+        start_cube (iris.cube.Cube):
             Cube containing the data at the beginning.
-        cube_1 (iris.cube.Cube):
+        end_cube (iris.cube.Cube):
             Cube containing the data at the end.
         interval_in_mins (int):
```

Comment:
We use SI units everywhere, so should this be in seconds? Could be then just `interval`.

Reply:
(Will pick this up/when doing the PR back into `master` - we can look at all of them and check they're consistent, I think).